src/parser.rs

```diff
@@ -1523,12 +1592,120 @@ pub fn guess_buffer_name(buffer: &[u8]) -> &str {
     while eol > index && (buffer[eol - 1] == b'\r' || buffer[eol - 1] == b' ') {
         eol -= 1;
     }
+	&buffer[index..eol]
+}
+
+/// Extract the parser commands out of a $j "additional information" comment
+fn commands(buffer: &[u8], pos: FilePos) -> Result<CommandIter, Diagnostic> {
+	let mut iter = CommandIter { buffer, index: pos as usize };
+	iter.skip_white_spaces();
+	iter.expect(b'$')?;
+	iter.expect(b'(')?;
+	iter.skip_white_spaces();
+	iter.expect(b'$')?;
+	iter.expect(b'j')?;
```

Comment:
What's going on with this iterator? It seems to be kind of useless, given the granularity of its output, compared to just using the `&[u8]` directly.

Reply:
`CommandIter` is parsing the `$j` commands and returning the command and arguments as a list of tokens. It's a facility for the later stages. 
Indeed, we could also defer this treatment to a later stage. I chose simplicity here because such comments are usually very scarce compared to the size of the whole database.

Reply:
The way I have done this in the past is that there is a top level MM statement iterator, which returns any comments that appear at statement level as separate items. For a comment, it will parse the `$(` up to the `$)` and give back the byte slice in between. Then this is passed to a comment parser, which checks for special comments (by using the split-on-space tokenizer to check if the next token is `$j`), and if so the remainder of the slice is parsed with the `$j` parser (which uses tokens like identifiers, strings and `;`).

Reply:
I think the comment parser would be [`get_comment`](https://github.com/david-a-wheeler/metamath-knife/blob/c4cf208547b019f1ad993efa6e2084bbd028a7bb/src/parser.rs#L991). 

I tried to stick to the same philosophy as the other results in [the `Segment` structure](https://github.com/david-a-wheeler/metamath-knife/blob/c4cf208547b019f1ad993efa6e2084bbd028a7bb/src/parser.rs#L427-L457). 

It contains parser results in a form which is usable by other parts of the program.

Reply:
I wonder whether it would be better for `get_comment` to record the start of the body of the comment after the `$j`/`$t`, for example by leveraging `Statement.proof_start` and `Statement.proof_end` (which is probably currently unused for comments). That way you don't have to deal with the fact that the parser of the first few characters of the comment is different from the rest.